engine/src/faction_generic.cpp

```diff
@@ -18,14 +22,16 @@ void Faction::ParseAllAllies()
 void Faction::ParseAllies( unsigned int thisfaction )
 {
     unsigned int i, j;
+    std::vector< faction_stuff >faction;
```

Comment:
OK, so I think what's happening here is that you're redeclaring this variable. I'm not sure that is the correct fix.

Reply:
I fully agree with you. Why would the compiler not see the previous declaration, though?

Reply:
I think the compiler is seeing the previous **declaration**, just not a previous **initialization with a value.** I think I'll take a look at the code and see if the compiler is right about that.

Reply:
Looks like gcc 10.x is complaining that [this field](https://github.com/vegastrike/Vega-Strike-Engine-Source/blob/7934960a2fa32a1ec1311c5fb6490c250df10d4f/engine/src/faction_generic.h#L33) is not initialized in [this constructor](https://github.com/vegastrike/Vega-Strike-Engine-Source/blob/7934960a2fa32a1ec1311c5fb6490c250df10d4f/engine/src/faction_generic.h#L37). Oddly enough, I don't get this warning in either gcc 9.3.0 or clang 10.0.0.

Try changing faction_generic.h, line 37, to the following:

```C++
        faction_stuff() : relationship(0.0)
        {
            stats.index = 0;
        }
```

See if that takes care of it.

Reply:
Indeed it did. Learn something new every day!